nsist/pypi.py

```diff
@@ -250,12 +252,83 @@ def extract_wheel(whl_file, target_dir, exclude=None):
     shutil.rmtree(str(td))
 
 
-def fetch_pypi_wheels(requirements, target_dir, py_version, bitness,
-                      extra_sources=None, exclude=None):
-    for req in requirements:
+def fetch_pypi_wheels(wheels_requirements, wheels_paths, target_dir, py_version,
+                      bitness, extra_sources=None, exclude=None):
+    """
+    Gather wheels included explicitly by wheels_pypi parameter 
+    or matching glob paths given in local_wheels parameter.
+    """
+    wheel_info_array = []
+    # We try to get the wheels from wheels_pypi requirements parameter
+    for req in wheels_requirements:
         wl = WheelLocator(req, py_version, bitness, extra_sources)
-        whl_file = wl.fetch()
+        whl_file = wl.fetch() 
+        validate_wheel(whl_file, wheel_info_array, py_version)
         extract_wheel(whl_file, target_dir, exclude=exclude)
+    # Then from the local_wheels paths parameter
+    for glob_path in wheels_paths:
+        for path in glob.glob(glob_path):
+            logger.info('Include wheel: %s (local_wheels path: %s)', os.path.basename(path), glob_path)
+            validate_wheel(path, wheel_info_array, py_version)
+            extract_wheel(path, target_dir, exclude=exclude)
+
+
+def validate_wheel(whl_path, wheel_info_array, py_version):
+    """
+    Verify that the given wheel can safely be included in the current installer.
```

Comment:
There's actually already code to do this in `WheelLocator` - see the `pick_best_wheel` method. It may need a bit of refactoring to use for this purpose, but I'd rather avoid duplicating the logic to check wheel tags.